src/shared/logger/winstonToolkitLogger.ts

```diff
@@ -0,0 +1,95 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as moment from 'moment'
+import * as vscode from 'vscode'
+import * as winston from 'winston'
+import { Loggable } from './loggableType'
+import { Logger, LogLevel } from './logger'
+import { OutputChannelTransport } from './outputChannelTransport'
+
+function formatMessage(level: LogLevel, message: Loggable[]): string {
+    // TODO : Look into winston custom formats - https://github.com/winstonjs/winston#creating-custom-formats
+    let final: string = `${makeLogTimestamp()} [${level.toUpperCase()}]:`
+    for (const chunk of message) {
+        if (chunk instanceof Error) {
+            final = `${final} ${chunk.stack}`
+        } else {
+            final = `${final} ${chunk}`
+        }
+    }
+
+    return final
+}
+
+function makeLogTimestamp(): string {
+    return moment().format('YYYY-MM-DD HH:mm:ss')
+}
+
+export class WinstonToolkitLogger implements Logger, vscode.Disposable {
+    // forces winston to output only pre-formatted message
+    private static readonly LOG_FORMAT = winston.format.printf(({ message }) => {
```

Comment:
What happens if we don't have this?

Reply:
Unsure. This is moving around existing code.